File: challenges/fibonnacci_iterative.rb
# Write a function, fibs(num) which returns the first n elements from
# the fibonnacci sequence, given n.

def fibs(num)
  return [] if num == 0
  return [0] if num == 1

  fibs = [0, 1]
  
  while fibs.count < num
    fibs << fibs[-1] + fibs[-2]
  end

  fibs
end

File: challenges/fibonnacci_iterative.js
// # Write a function, fibs(num) which returns the first n elements from
// # the fibonnacci sequence, given n.

function fibs(num) {
  if (num === 0) {
    return [];
  } else if (num === 1) {
    return [0];
  }

  let result = [0, 1];

  while (result.length < num) {
    result.push(result[result.length - 1] + result[result.length - 2]);
  }

  return result;
}
